File: code/community/1304604967260131333/deepgram-rttm-file-format-diarization.go
package main

import (
	"bytes"
	"encoding/json"
	"fmt"
	"io/ioutil"
	"net/http"
	"os"
)

func main() {
	apiKey := os.Getenv("DEEPGRAM_API_KEY")
	audioUrl := "https://your-audio-file-url.com/audio.wav"

	client := &http.Client{}
	
	requestBody, err := json.Marshal(map[string]interface{}{
		"url":       audioUrl,
		"diarize":   true,
		"punctuate": true,
	})
	if err != nil {
		panic(err)
	}

	req, err := http.NewRequest("POST", "https://api.deepgram.com/v1/listen", bytes.NewBuffer(requestBody))
	if err != nil {
		panic(err)
	}

	req.Header.Set("Authorization", fmt.Sprintf("Token %s", apiKey))
	req.Header.Set("Content-Type", "application/json")

	resp, err := client.Do(req)
	if err != nil {
		panic(err)
	}
	defer resp.Body.Close()

	body, err := ioutil.ReadAll(resp.Body)
	if err != nil {
		panic(err)
	}

	fmt.Println(string(body))
}

File: code/community/1304604967260131333/deepgram-rttm-file-format-diarization.cs
using System;
using System.Net.Http;
using System.Threading.Tasks;

class Program
{
    private static readonly HttpClient client = new HttpClient();

    static async Task Main()
    {
        string apiKey = Environment.GetEnvironmentVariable("DEEPGRAM_API_KEY");
        string audioUrl = "https://your-audio-file-url.com/audio.wav";

        client.DefaultRequestHeaders.Add("Authorization", $"Token {apiKey}");

        var requestBody = new {
            url = audioUrl,
            diarize = true,
            punctuate = true
        };

        var response = await client.PostAsJsonAsync(
            "https://api.deepgram.com/v1/listen",
            requestBody
        );

        string responseString = await response.Content.ReadAsStringAsync();
        Console.WriteLine(responseString);
    }
}